lxd/storage_pools_utils.go

```diff
@@ -220,54 +224,90 @@ func storagePoolCreateInternal(state *state.State, poolName, poolDescription str
 		if !tryUndo {
 			return
 		}
-		dbStoragePoolDeleteAndUpdateCache(state.Cluster, poolName)
+
+		dbStoragePoolDeleteAndUpdateCache(state.Cluster, req.Name)
 	}()
-	err = doStoragePoolCreateInternal(state, poolName, poolDescription, driver, config, false)
-	tryUndo = err != nil
-	return err
-}
 
-// This performs all non-db related work needed to create the pool.
-func doStoragePoolCreateInternal(state *state.State, poolName, poolDescription string, driverName string, config map[string]string, isNotification bool) error {
-	tryUndo := true
-	s, err := storagePoolInit(state, poolName)
+	err = storagePoolCreateLocal(state, id, req, false)
 	if err != nil {
 		return err
 	}
 
-	// If this is a clustering notification for a ceph storage, we don't
-	// want this node to actually create the pool, as it's already been
-	// done by the node that triggered this notification. We just need to
-	// create the storage pool directory.
-	if s, ok := s.(*storageCeph); ok && isNotification {
-		volumeMntPoint := driver.GetStoragePoolVolumeMountPoint(s.pool.Name, s.volume.Name)
-		return os.MkdirAll(volumeMntPoint, 0711)
+	tryUndo = false
+	return nil
+}
 
-	}
-	err = s.StoragePoolCreate()
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if !tryUndo {
-			return
+// This performs all non-db related work needed to create the pool.
+func storagePoolCreateLocal(state *state.State, id int64, req api.StoragePoolsPost, isNotification bool) error {
+	tryUndo := true
+
+	// Make a copy of the req for later diff.
+	var updatedConfig map[string]string
+	var updatedReq api.StoragePoolsPost
+	shared.DeepCopy(&req, &updatedReq)
+
+	// Attempt to create using the new storage pool logic.
+	pool, err := storagePools.CreatePool(state, id, &updatedReq, isNotification, nil)
```

Comment:
@stgraber could the DB record creation for the pool be moved into `CreatePool` then we wouldn't need the logic to detect changes and update the database again?